src/solvers/refinement/string_constraint_generator_float.cpp

```diff
@@ -0,0 +1,445 @@
+/*******************************************************************\
+
+Module: Generates string constraints for functions generating strings
+        from other types, in particular int, long, float, double, char, bool
+
+Author: Romain Brenguier, romain.brenguier@diffblue.com
+
+\*******************************************************************/
+
+/// \file
+/// Generates string constraints for functions generating strings from other
+/// types, in particular int, long, float, double, char, bool
+
+#include <solvers/floatbv/float_bv.h>
+
+#include "string_constraint_generator.h"
+
+
+/// Gets the unbiased exponent in a floating-point bit-vector
+///
+/// TODO: factorize with float_bv.cpp float_utils.h
+/// \param src: a floating point expression
+/// \param spec: specification for floating points
+/// \return A 32 bit integer representing the exponent. Note that 32 bits are
+///   sufficient for the exponent even in octuple precision.
```

Comment:
for strings I usually say `\return a new string expression` so here I could say `\return A new 32 bit integer expression`